core/iwasm/compilation/aot_compiler.c

```diff
@@ -146,9 +146,20 @@ aot_validate_wasm(AOTCompContext *comp_ctx)
     }
 
 #if WASM_ENABLE_MEMORY64 != 0
-    if (comp_ctx->pointer_size < sizeof(uint64) && IS_MEMORY64) {
-        aot_set_last_error("Compiling wasm64 to 32bit platform is not allowed");
-        return false;
+    if (comp_ctx->pointer_size < sizeof(uint64)) {
+        if (IS_MEMORY64) {
```

Comment:
I'm just wondering, why does L150 only check `memories[0]` while L157 needs to go through the entire table, considering there could be multiple memories?

Reply:
Currently, we assume that the memory is all 64bit or all 32bit, so it's okay only to check the first memory section